apps/server/src/modules/task/uc/task.uc.ts

```diff
@@ -56,18 +58,26 @@ export class TaskUC {
 			{ pagination, order: { dueDate: SortOrder.desc } }
 		);
 
-		const taskWithStatusVos = tasks.map((task) => {
+		const taskWithStatusVos = tasks.map(async (task) => {
 			let status: ITaskStatus;
 			if (this.authorizationService.hasPermission(user, task, PermissionContextBuilder.write([]))) {
 				status = task.createTeacherStatusForUser(user);
+				if (task.taskCard) {
+					status.taskCardCompleted = await this.taskCardService.countCompletedForUsers(user.id, task.taskCard);
+				}
 			} else {
 				status = task.createStudentStatusForUser(user);
+				if (task.taskCard) {
+					status.isTaskCardCompleted = await this.taskCardService.isCompletedForUser(user.id, task.taskCard);
+				}
```

Comment:
can this be moved in createTeacherStatusForuser and createStudentStatusForUser ? 
it's confusing to have a function calculating the status, and then extend it or in many other places

Reply:
that was my plan in the begining, but `createTeacherStatusForUser` and student are calculated in the task entity. And as we only have the taskCardId there as a string instead of the taskCard entity, we can not get the information there (otherwise we might not need the task card service I guess). 

So its either we get the entity there in task entity (which might lead to performance issues?) or we separate it and we have additionally function calls here in the UC and in task service. Lets discuss on monday